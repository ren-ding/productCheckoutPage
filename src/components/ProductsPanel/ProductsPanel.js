import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class ProductsPanel extends Component {
    render() {
        return (
            <div>ProductsPanel</div>
        );
    }
}

ProductsPanel.propTypes = {
    products: PropTypes.array.isRequired
};