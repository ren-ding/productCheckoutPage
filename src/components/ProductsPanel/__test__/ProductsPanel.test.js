import React from 'react';
import ProductsPanel from '../ProductsPanel';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

describe('ProductsPanel', ()=> {
    let productsPanel;
    let props;

    beforeEach(()=> {
        props = {
            products: []
        }

        productsPanel = () => shallow(<ProductsPanel {...props} />, { lifecycleExperimental: true });        
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<ProductsPanel {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });
});