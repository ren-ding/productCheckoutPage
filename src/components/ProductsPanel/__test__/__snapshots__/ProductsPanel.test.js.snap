// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPanel render should render to match snapshot 1`] = `
<div>
  ProductsPanel
</div>
`;
